backend/src/main/java/fk/prof/backend/service/ProfileWorkService.java

```diff
@@ -0,0 +1,41 @@
+package fk.prof.backend.service;
+
+import fk.prof.backend.aggregator.AggregationWindow;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+//TODO: Liable for refactoring. For now, placeholder to enable functional completion of /profile api
+public class ProfileWorkService implements IProfileWorkService {
+  private AggregationWindowStore aggregationWindowStore = new AggregationWindowStore();
+
+  public void associateAggregationWindow(Long workId, AggregationWindow aggregationWindow) {
+    this.aggregationWindowStore.add(workId, aggregationWindow);
+  }
+
+  public AggregationWindow getAssociatedAggregationWindow(Long workId) {
+    return this.aggregationWindowStore.get(workId);
+  }
+
+  /**
+   * TODO: Evaluate if this is the right DS. Temporary for now
+   * Ideally, there should be separate in-memory store/persistent store impl for storing work-id related data model
+   */
+  public static class AggregationWindowStore {
+
+    private ConcurrentHashMap<Long, AggregationWindow> windowLookup = new ConcurrentHashMap<>();
+
+    public void add(long workId, AggregationWindow window) {
+      this.windowLookup.putIfAbsent(workId, window);
+    }
+
+    public AggregationWindow get(long workId) {
+      return this.windowLookup.get(workId);
+    }
+
+    public void removeWorkIds(long[] workIds) {
```

Comment:
So I have kept around a bunch of such functions because I see them immediately getting used when aggregation window expiry related code comes in. Another example is abandonProfile method in AggregationWindow class. Deleting them right now, only to add them a week later (mostly by the same dev) does not make sense. Of course, relevant test for these will be added later so there is a small risk of incorrect code being merged early but it should be anyways caught when we write the classes using this method.